doc/developer_guide/psyir_backends.rst

```diff
@@ -332,25 +332,34 @@ Back-ends for the PSy-layer
 
 The additional complexity of the PSy-layer comes from the fact that it
 contains multiple domain-specific concepts and parallel concepts that are not
-part of the target languages. The purpose of the Visitors are to encapsulate
-in a single place the translation logic from the generic PSyIR constructs into
-the target language. So it seems appropriate that these additional
-domain-specific generation knowledge (which is also common between backends)
-shouldn't belong to the Visitor, but in the API-specific nodes themself.
-
-The current proposed solution is create a 2-phase generation workflow where
+part of the target languages. Instead of dealing with this concepts in the
+visitors we require that any domain-specific concept introduced in top
+of the core PSyIR constructs contains the logic to lower this concept
+into language level constructs. The reasons to choose a method instead
+of a visitor for this transformation are:
+
+- Each concept introduced by the API-developer will need lowering instructions,
+  and this is better implied by an abstract class in the node that needs to
+  be filled.
+- It fits better with the in-place transformation currently used for the
```

Comment:
I rephrased this sentence.

To be clear I think the method is the best solution here. If the text sounded hesitant is because this is a work in progress and I didn't want to sound completely categorical. But I guess this is implied by the "Under development" note below.

Reply:
Aha, I get it now - thanks.